lib/dfe/analytics/entity_processor.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'shared/checksum_logic'
+
+module DfE
+  module Analytics
+    # Processes entities by sending the import_entity_check_event
+    class EntityProcessor
```

Comment:
The `EntityProcessor` and the `ChecksumLogic` is factoring out the common logic and is a step in the right direction. I think we should go a step further though and create a service object for the table check logic.

Most of the rails projects use the following service pattern:

``` Ruby
module ServicePattern
  def self.included(base)
    base.extend(ClassMethods)
    base.class_eval do
      private_class_method(:new)
    end
  end

  def call
    raise(NotImplementedError("#call must be implemented"))
  end

  module ClassMethods
    def call(*args, **keyword_args, &block)
      return new.call if args.empty? && keyword_args.empty? && block.nil?

      new(*args, **keyword_args, &block).call
    end
  end
end
```

I won't go into the details here as it's too much typing, but we can have a chat about it.

Reply:
Fully onboard!